widget/embedded/src/containers/WidgetProvider/WidgetProvider.tsx

```diff
@@ -12,13 +13,21 @@ import { WidgetInfo } from '../WidgetInfo';
 export function WidgetProvider(props: PropsWithChildren<PropTypes>) {
   const { onUpdateState, config } = props;
 
-  useMemo(() => {
+  useEffect(() => {
```

Comment:
Why you've changed memo to effect?

Reply:
To improve the code quality. That was a wrong way of using 'useMemo'.

Reply:
They are running in 2 different phases, one in render and effect after a render in commit phase. This could break the behavior here. Did you check it?
Although I agree useMemo is for caching a result so maybe it's a misuse here.

Reply:
Yes I think it is a misuse of useMemo and it should be corrected. Also the effect of 'initConfig' with apiKey and apiUrl is not necessary for render phase and it can be happen after the render. The only unintended behavior can be api calls with the wrong apiKey and apiUrl. I tested that and didn't detect this unintended behavior. Considering this, also I don't think we need to use some alternative like useLayoutEffect here.

Reply:
I will leave this to @Ikari-Shinji-re to check and resolve this comment.

Reply:
Considering that 'API_KEY' and 'BASE_URL' are not needed during the render phase, and no memoization is happened here I think it wont have any side effect. check please @Ikari-Shinji-re